com.kms.cura.dal/src/com/kms/cura/dal/database/FacilityDatabaseHelper.java

```diff
@@ -0,0 +1,46 @@
+package com.kms.cura.dal.database;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.kms.cura.dal.mapping.EntityColumn;
+import com.kms.cura.dal.mapping.FacilityColumn;
+import com.kms.cura.entity.DayOfTheWeek;
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.FacilityEntity;
+import com.kms.cura.entity.OpeningHour;
+
+public class FacilityDatabaseHelper extends DatabaseHelper {
+
+	public FacilityDatabaseHelper() throws ClassNotFoundException, SQLException {
+		super();
+	}
+
+	@Override
+	protected Entity getEntityFromResultSet(ResultSet resultSet) throws SQLException, ClassNotFoundException {
+		List<OpeningHour> openingHours = new ArrayList<OpeningHour>();
+		ResultSet openingHoursRS = null;
+		try {
+			openingHoursRS = this.queryByReferenceID("opening_hours", "facility_id",
+					resultSet.getInt(EntityColumn.ID.getColumnName()));
+			while (openingHoursRS.next()) {
+				openingHours.add(new OpeningHour(DayOfTheWeek.getDayOfTheWeek(openingHoursRS.getInt("week_day")),
+						openingHoursRS.getTime("time_open"), openingHoursRS.getTime("time_close")));
```

Comment:
Should put opening_hours, facility_id, week_day, time_open, time_close into OpeningHourColumn